src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -0,0 +1,376 @@
+ */
+    BOOST_AUTO_TEST_CASE(KmClusteringSizeCheckLabelOnly)
+    {
+        constexpr int N = 10;
+        constexpr int D = 4;
+        int C = 2;
+
+        arma::mat InputData = arma::randu<arma::mat>(N, D);
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("labels_only", true);
+
+
+        mlpackMain();
+
+        BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_rows, N);
+        BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_cols, 1);
+        BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_rows, C);
+        BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_cols, D);
+    }
+
+
+/**
+ * Checking that predictions are not same when --allow_empty_clusters or kill_empty_clusters are specified
+ */
+    BOOST_AUTO_TEST_CASE(KmClusteringEmptyClustersCheck)
+    {
+        constexpr int N = 100;
+        constexpr int D = 4;
+        int C = 95;
+        int iterations = 100;
+        arma::mat InputData = arma::randu<arma::mat>(N, D);
+        arma::mat InitialCentroids=arma::ones(C, D);
+
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("labels_only", true);
+        SetInputParam("initial_centroids", std::move(InitialCentroids));
+        SetInputParam("max_iterations", std::move(iterations));
+
+        mlpackMain();
+
+        arma::mat NormalOutput;
+        NormalOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+        ResetKmSettings();
+
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("labels_only", true);
+        SetInputParam("initial_centroids", std::move(InitialCentroids));
+        SetInputParam("allow_empty_clusters", true);
+        SetInputParam("max_iterations", std::move(iterations));
+
+        mlpackMain();
+
+        arma::mat AllowEmptyOutput;
+        AllowEmptyOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+        ResetKmSettings();
+
+        SetInputParam("input", std::move(InputData));
+        SetInputParam("clusters", std::move(C));
+        SetInputParam("labels_only", true);
+        SetInputParam("initial_centroids", std::move(InitialCentroids));
+        SetInputParam("kill_empty_clusters", true);
+        SetInputParam("max_iterations", std::move(iterations));
+
+        mlpackMain();
+
+        arma::mat KillEmptyOutput;
+        KillEmptyOutput = std::move(CLI::GetParam<arma::mat>("output"));
+
+        ResetKmSettings();
+
+        //Checking that the results are not identical
+
+        for (int j = 0; j <N ; ++j)
+        {
+            BOOST_CHECK_NE(NormalOutput(j), AllowEmptyOutput(j));
+        }
+        for (int j = 0; j <N ; ++j)
+        {
+            BOOST_CHECK_NE(KillEmptyOutput(j), AllowEmptyOutput(j));
+        }
```

Comment:
These checks will ensure that every single element in the output is different, but that's not necessarily what we require---we only need that a single element in the output differs.  It's possible that the results could be different, but one single element of `NormalOutput` or `KillEmptyOutput` or `AllowEmptyOutput` could be the same. Can you adapt this please?